akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala

```diff
@@ -747,31 +772,34 @@ class JdbcProjectionSpec
       projectionTestKit.runWithTestSink(atLeastOnceProjection(_ == 4)) { sinkProbe =>
         sinkProbe.request(3)
         eventually {
-          projectedValueShouldBe("abc|def|ghi")
-          // we are saving after each 2 envelopes!
-          // Offset 3 won't be saved because `afterDuration` is 1.minute
-          offsetShouldBe(2)
+          projectedValueShouldBe("e1|e2|e3")
         }
       }
 
       // run again up to failure point
       projectionTestKit.runWithTestSink(atLeastOnceProjection(_ == 4)) { sinkProbe =>
-        sinkProbe.request(2) // processes elem 3 again and fails when processing 4
+        sinkProbe.request(4) // processes elem 3 again and fails when processing 4
         eventuallyExpectError(sinkProbe).getMessage should startWith(concatHandlerFail4Msg)
         eventually {
-          projectedValueShouldBe("abc|def|ghi|ghi") // note elem 3 is processed twice
-          offsetShouldBe(2)
+          // because failures, we may consume 'e1' and 'e2' more then once
+          // we check that it at least starts with 'e1|e2|e3'
+          projectedValueShould(_.startsWith("e1|e2|e3"))
+
+          // note elem 3 is processed twice
+          projectedValueShouldIncludeNTimes("e3", 2)
         }
       }
 
       // re-run projection without failing function
-      val projection = atLeastOnceProjection()
+      projectionTestKit.run(atLeastOnceProjection()) {
+        projectedValueShould(_.startsWith("e1|e2|e3"))
+        projectedValueShould(_.endsWith("e3|e4|e5|e6"))
 
-      projectionTestKit.run(projection) {
-        // note that 3rd is triplicate
-        projectedValueShouldBe("abc|def|ghi|ghi|ghi|jkl|mno|pqr")
+        // note elem 3 is should have been seen three times
+        projectedValueShouldIncludeNTimes("e3", 3)
+
+        offsetShouldBe(6L)
       }
-      offsetShouldBe(6L)
     }
 
```

Comment:
As discussed with @renatocaval, this change shouldn't be necessary and still running the `offsetShouldBe(6L)`  out of the `projectionTestKit.run` caused an execution failure. 

We'll merge anyway to unblock other in-flight PRs and eventually com back to this is the tests still proves to be flaky.